src/gt4py/next/common.py

```diff
@@ -95,6 +94,11 @@ def __post_init__(self):
             object.__setattr__(self, "start", 0)
             object.__setattr__(self, "stop", 0)
 
+    @classmethod
+    @property
+    def infinity(cls) -> UnitRange:
```

Comment:
might need a `#type: ignore[xxx]` because classmethod+property is a python 3.9 feature

Reply:
maybe also make the `Infinity.positive/negative` a property?